src/main/k8s/00-main.tf

```diff
@@ -37,4 +41,15 @@ provider "kubernetes" {
   token                  = data.aws_eks_cluster_auth.this.token
 }
 
+provider "helm" {
+  kubernetes {
+    host                   = data.aws_eks_cluster.this.endpoint
+    cluster_ca_certificate = base64decode(data.aws_eks_cluster.this.certificate_authority[0].data)
+    exec {
+      api_version = "client.authentication.k8s.io/v1beta1"
+      args        = ["eks", "get-token", "--cluster-name", data.aws_eks_cluster.this.name]
+      command     = "aws"
+    }
```

Comment:
Can you test if it works by using `token` instead? See the `kubernetes` provider configuration (line 38)